pandas/core/internals/construction.py

```diff
@@ -285,13 +285,16 @@ def extract_index(data):
         have_raw_arrays = False
         have_series = False
         have_dicts = False
+        have_ordered = False
 
         for val in data:
             if isinstance(val, ABCSeries):
                 have_series = True
                 indexes.append(val.index)
             elif isinstance(val, dict):
                 have_dicts = True
+                if isinstance(val, OrderedDict):
+                    have_ordered = True
```

Comment:
Just set ``sort=False`` instead? Also, I don`t get why OrderedDict is special: a regular dict is also ordered from python 3.6.

Reply:
Thank you for reviewing. I agree with you (a dict is also ordered). 
I implemented this way according to [this comment](https://github.com/pandas-dev/pandas/issues/8425#issuecomment-57616916). But now I understand that we don't need to differentiate between a dict and an OrderedDict anymore. I'll fix my codes.

Reply:
yeah that comment is pretty old
the sort parameter was recently added

Reply:
@jreback, thank you.